### PR TITLE
Upgrade to openssl3, gtest 1.14 and vs2022 c++20.

### PR DESCRIPTION
**Pull request by Thymir**
Upgrade to openssl 3.0.12
Upgrade to VS 2022
Upgrade to Gtest 1.14
Upgrade to c++ 20

Combine pipeline functionalities.

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -11,16 +11,15 @@ class HashUtilsConan(ConanFile):
...
 
 	def configure(self):
 		self.options["openssl"].shared = True
 
 	def requirements(self):
-		self.requires("openssl/1.1.1w#2d78cf15871a2cd0bc206c3d560cf741")
-		self.requires("gtest/1.10.0", private=True)
+		self.requires("zlib/1.2.13#13c96f538b52e1600c40b88994de240f", override=True)
+		self.requires("openssl/3.0.12#1670458f93ec138c3bb6afc65a1cd667")
+		self.requires("gtest/1.14.0#4372c5aed2b4018ed9f9da3e218d18b3", private=True)
```
Comment (franpuerta) [inline conanfile.py:22]: I `gtest` also required in this conanfile?

**Review by franpuerta — APPROVED** [review]

File: ci/pipelines/Build.jenkinsfile [hunk 2]
```diff
@@ -18,7 +17,15 @@ pipeline
 
 	parameters
 	{
-		booleanParam( name: 'uploadTestingPkg',
+		string(	name: 'version',
+				description: 'Version to build (must match the name of the tag that will be checked out), leave blank for checkout of current branch',
+				defaultValue: '' )
```
Comment (Blackvals) [inline ci/pipelines/Build.jenkinsfile]: Why not with 0.0.0 which is the default version value that will be used when building?

**Review by Blackvals — APPROVED** [review]

File: ci/pipelines/Build.jenkinsfile (hunk 2 above)
Comment (Thymir) [reply ci/pipelines/Build.jenkinsfile]: When I refactor a pipeline I typically try to leave the version default as I encountered it. This was just maintaining the same philosophy. But I agree, I'll change it to 0.0.0. It makes more sense as no tag 0.0.0 exists.

File: conanfile.py (hunk 1 above)
Comment (Thymir) [reply conanfile.py:22]: This is the "We don't know enough cmake solution" We need gtest for the test project inside the solution. As gtest doesn't apply to HashUtils is set as private so is not reported to users.

**Pull request merged**